src/backend/logger/__tests__/logfile.test.ts

```diff
@@ -0,0 +1,186 @@
+import { DirResult, dirSync } from 'tmp'
+import graceful_fs from 'graceful-fs'
+import { join } from 'path'
+import { app } from 'electron'
+import { configStore } from '../../constants'
+import * as constants from '../../constants'
+import * as logfile from '../logfile'
+import { logError } from '../logger'
+
+jest.mock('electron')
+jest.mock('electron-store')
+jest.mock('../../constants')
+jest.mock('../logger')
+jest.unmock('../logfile')
+
+let tmpDir = {} as DirResult
+
+describe('logger/logfile.ts', () => {
+  beforeEach(() => {
+    tmpDir = dirSync({ unsafeCleanup: true })
+  })
+
+  afterEach(() => {
+    jest.restoreAllMocks()
+    tmpDir.removeCallback()
+  })
+
+  test('createNewLogFileAndClearOldOnces fails because logDir does not exist', () => {
+    const spyAppGetPath = jest.spyOn(app, 'getPath').mockReturnValue('invalid')
+    const spyOpenSync = jest.spyOn(graceful_fs, 'openSync')
+
+    logfile.createNewLogFileAndClearOldOnces()
+
+    expect(spyOpenSync).toBeCalledWith(
+      expect.stringContaining('invalid/heroic-'),
+      'w'
+    )
+    expect(spyAppGetPath).toBeCalledWith('logs')
+    expect(logError).toBeCalledWith(
+      [
+        expect.stringContaining(`Open invalid/heroic-`),
+        expect.objectContaining(
+          Error("ENOENT: no such file or directory, open 'invalid/heroic-")
+        )
+      ],
+      { prefix: 'Backend', skipLogToFile: true }
+    )
+  })
+
+  test('createNewLogFileAndClearOldOnces success', () => {
+    jest.spyOn(app, 'getPath').mockReturnValue(tmpDir.name)
+    jest.spyOn(configStore, 'has').mockReturnValue(true)
+    jest.spyOn(configStore, 'get').mockReturnValue({
+      currentLogFile: 'old/log/path/file.log',
+      lastLogFile: undefined
+    })
+
+    const data = logfile.createNewLogFileAndClearOldOnces()
+
+    expect(logError).not.toBeCalled()
+    expect(data).toStrictEqual({
+      currentLogFile: expect.any(String),
+      lastLogFile: 'old/log/path/file.log'
+    })
+  })
+
+  test('createNewLogFileAndClearOldOnces removing old logs fails', () => {
+    jest.spyOn(app, 'getPath').mockReturnValue(tmpDir.name)
+    const spyUnlinkSync = jest
+      .spyOn(graceful_fs, 'unlinkSync')
+      .mockImplementation(() => {
+        throw Error('unlink failed')
+      })
+    const date = new Date()
+    date.setMonth(date.getMonth() > 0 ? date.getMonth() - 1 : 11)
+    const monthOutdated = join(
+      tmpDir.name,
+      // @ts-ignore replaceAll error
+      `heroic-${date.toISOString().replaceAll(':', '_')}.log`
```

Comment:
I have no clue. I already tried to setup a new tsconfig for the tests. I feel like vscode does something wrong and does not apply tsconfig to the test files, because everything works fine.